src/box/tuple_bloom.c

```diff
@@ -469,27 +470,20 @@ tuple_bloom_decode(const char **data, enum tuple_bloom_version version)
 			unreachable();
 		if (mp_decode_uint(data) != 0) /* version */
 			unreachable();
-		bloom->parts[0].table_size = mp_decode_uint(data);
-		bloom->parts[0].hash_count = mp_decode_uint(data);
+		bloom->parts[0].bloom.table_size = mp_decode_uint(data);
+		bloom->parts[0].bloom.hash_count = mp_decode_uint(data);
 		size_t store_size = mp_decode_binl(data);
-		assert(store_size == bloom_store_size(&bloom->parts[0]));
-		if (bloom_load_table(&bloom->parts[0], *data) != 0) {
-			diag_set(OutOfMemory, store_size, "bloom_load_table",
-				 "tuple bloom part");
-			free(bloom);
-			return NULL;
-		}
+		assert(store_size ==
+		       bloom_data_size(&bloom->parts[0].bloom));
+		bloom->parts[0].data = xmalloc(store_size);
+		memcpy(bloom->parts[0].data, *data, store_size);
 		*data += store_size;
 		break;
 	case TUPLE_BLOOM_VERSION_V2:
 	case TUPLE_BLOOM_VERSION_V3:
 		bloom->part_count = 0;
 		for (uint32_t i = 0; i < part_count; i++) {
-			if (tuple_bloom_decode_part(&bloom->parts[i],
-						    data) != 0) {
-				tuple_bloom_delete(bloom);
-				return NULL;
-			}
+			tuple_bloom_decode_part(&bloom->parts[i], data);
```

Comment:
Let's please make `tuple_bloom_decode` non-failing too - we just need to replace the `malloc` with `xmalloc` in the beginning of this function.